libtransact/src/contract/context/key_value.rs

```diff
@@ -0,0 +1,1310 @@
+                }
+            });
+            Ok(())
+        }
+
+        fn delete_entries(&mut self, addresses: &[String]) -> Result<Vec<String>, ContextError> {
+            let mut deleted = Vec::new();
+            addresses.iter().for_each(|key| {
+                if let Some(_) = self.state.remove(key) {
+                    deleted.push(key.to_string());
+                }
+            });
+            Ok(deleted)
+        }
+
+        fn add_receipt_data(&self, _data: Vec<u8>) -> Result<(), ContextError> {
+            Ok(())
+        }
+
+        fn add_event(
+            &self,
+            _event_type: String,
+            _attributes: Vec<(String, String)>,
+            _data: Vec<u8>,
+        ) -> Result<(), ContextError> {
+            Ok(())
+        }
+    }
+
+    /// TestContext using the simple TestState backend to be used as the internal context to
+    /// construct a KeyValueTransactionContext for the tests.
+    struct TestContext {
+        internal_state: Arc<Mutex<TestState>>,
+    }
+
+    impl TestContext {
+        pub fn new() -> Self {
+            TestContext {
+                internal_state: Arc::new(Mutex::new(TestState::new())),
+            }
+        }
+    }
+
+    /// Implements the TransactionContext trait for the TestContext.
+    impl TransactionContext for TestContext {
+        fn get_state_entries(
+            &self,
+            addresses: &[String],
+        ) -> Result<Vec<(String, Vec<u8>)>, ContextError> {
+            self.internal_state
+                .lock()
+                .expect("Test lock was poisoned in get method")
+                .get_entries(addresses)
+        }
+
+        fn set_state_entries(&self, entries: Vec<(String, Vec<u8>)>) -> Result<(), ContextError> {
+            self.internal_state
+                .lock()
+                .expect("Test lock was poisoned in set method")
+                .set_entries(entries)
+        }
+
+        fn delete_state_entries(&self, addresses: &[String]) -> Result<Vec<String>, ContextError> {
+            self.internal_state
+                .lock()
+                .expect("Test lock was poisoned in delete method")
+                .delete_entries(addresses)
+        }
+
+        fn add_receipt_data(&self, data: Vec<u8>) -> Result<(), ContextError> {
+            self.internal_state
+                .lock()
+                .expect("Test lock was poisoned in add_receipt_data method")
+                .add_receipt_data(data)
+        }
+
+        fn add_event(
+            &self,
+            event_type: String,
+            attributes: Vec<(String, String)>,
+            data: Vec<u8>,
+        ) -> Result<(), ContextError> {
+            self.internal_state
+                .lock()
+                .expect("Test lock was poisoned in add_event method")
+                .add_event(event_type, attributes, data)
+        }
+    }
+
+    /// Create a HashMap from `key` and `value` to create a StateEntryValue message for a
```

Comment:
"Creates" (for consistency)

Reply:
Similar to above, reworked these comments to begin with a noun rather than a verb

Reply:
Reworded these comments to start with a noun